lib/ClangImporter/ImportName.cpp

```diff
@@ -2200,6 +2200,45 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
     }
   }
 
+  SmallString<16> newName;
+  // Check if we need to rename the C++ method to disambiguate it.
+  if (auto method = dyn_cast<clang::CXXMethodDecl>(D)) {
+    if (!method->isConst() && !method->isOverloadedOperator()) {
+      // See if any other methods within the same struct have the same name and
+      // parameters, but differ in constness.
+      auto otherDecls = dc->lookup(method->getDeclName());
+      bool shouldRename = false;
+      for (auto otherDecl : otherDecls) {
+        if (otherDecl == D)
+          continue;
+        if (auto otherMethod = dyn_cast<clang::CXXMethodDecl>(otherDecl)) {
+          // TODO: what if the other method is also non-const?
+          if (!otherMethod->isConst())
+            continue;
+          // If the parameters are different, there is no need to disambiguate.
+          if (otherMethod->getNumParams() != method->getNumParams())
+            continue;
+          bool sameParams = llvm::none_of(
```

Comment:
I don't think we were comparing parameter types before , were we? why are we comparing them now.

Reply:
You're right, thanks, I assumed for some reason that we were comparing `DeclName`s, but turns out it was just strings. I'll get rid of this check.